blocks/library/embed/index.js

```diff
@@ -118,17 +144,24 @@ registerBlockType( 'core/embed', {
 						} else {
 							this.setState( { error: true } );
 						}
-						this.setState( { fetching: false } );
+						this.setState( { fetching: false, loadingFromSavedBlock: false } );
 					} );
 				}
 			);
 		}
 
 		render() {
-			const { html, type, error, fetching } = this.state;
+			const { html, type, error, fetching, loadingFromSavedBlock } = this.state;
```

Comment:
I think we need @jasmussen to cast a design spell on the loading state so it's suitable for both purposes :smile: